### ISSUE TITLE
Media: implement image editing tools such as crop, rotate, flip, and scale

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-calypso#2098]
Raised by @aduth 

From the media detail view, a user has the option to toggle into an editable mode, which allows them to crop, rotate, flip, and scale an image.

![media-library-modal-edit](https://cloud.githubusercontent.com/assets/1779930/7323708/aaf18688-ea7e-11e4-96e7-7bd87393046f.png)

Also see p5j4vm-yo-p2

**Comment by rralian** [comment]
I'm moving this issue into the `plans iteration 1` milestone, because team triton will be interested/active followers here (and that's the milestone we're tracking). We think there's some opportunity to offer advanced image editing as part of plans, but we need basic image editing here first. :-)

@robobot3000 I saw you had created issue #5001 and then closed it as a duplicate of this. But I think there's value in copying over your notes from that issue if we discuss implementation here.

**Comment by robobot3000** [comment]
The mockups were updated, so I am uploading the new ones:

![group-16-copy](https://cloud.githubusercontent.com/assets/800604/14898696/fba8924a-0d7f-11e6-9c38-2430a47612cd.png)
![group-17-copy](https://cloud.githubusercontent.com/assets/800604/14898698/028c3cba-0d80-11e6-8a80-ce160a983f3b.png)

I am in the process of implementing the functionality using a canvas element, and there doesn't seem to be any need for new plugins.

**Comment by robobot3000** [comment]
Looks like the image upload from canvas is blocked by #5099. I will continue working on the UI while this is addressed. It should be possible to replace the canvas component with something else if needed.

CC'ing @nb, so he's aware of this blocker.

**Comment by robobot3000** [comment]
Just a note, if the canvas issue takes too long to solve, I can use Photon API when saving images.

**Comment by gwwar** [comment]
If the Photon API seems performant enough, lets go ahead with that solution for now.

**Comment by gwwar** [comment]
Let's close this one out, since @robobot3000 added this in #5218 and we have a milestone for the image editor
